rtengine/camconst.cc

```diff
@@ -606,6 +618,24 @@ CameraConst::get_WhiteLevel(const int idx, const int iso_speed, const float fnum
     return lvl.levels[idx];
 }
 
+bool
+CameraConst::has_globalGreenEquilibration()
+{
+    return globalGreenEquilibration >= 0;
```

Comment:
I know the `-1` "trick" is very common. A nicer solution will be [`std::optional<>`](http://en.cppreference.com/w/cpp/utility/optional) in C++17. In the meantime, we could add our own handy `Optional<>`:

```c++
template<class T>
class Optional
{
public:
    Optional() :
        present(false),
        value(T())
    {
    }

    Optional(const T& _value) :
        present(true),
        value(_value)
    {
    }

    ~Optional() = default;

    Optional(const Optional& other) = default;
    Optional& operator =(const Optional& other) = default;

    explicit operator bool() const
    {
        return present;
    }

    T getValue(const T& default_value) const
    {
        return
            present
                ? value
                : default_value;
    }

    const T& operator *() const
    {
        return value;
    }

    T& operator *()
    {
        return value;
    }

    const T* operator ->() const
    {
        return &value;
    }

    Optional& operator =(const T& new_value)
    {
        present = true;
        value = new_value;
        return *this;
    }

    void reset()
    {
        present = false;
        value = T();
    }

private:
    bool present;
    T value;
};
```